plugins/kotlin-dataframe/src/org/jetbrains/kotlinx/dataframe/plugin/impl/api/convert.kt

```diff
@@ -140,3 +152,31 @@ internal class To0 : AbstractInterpreter<PluginDataFrameSchema>() {
         return convertImpl(receiver.schema, receiver.columns, typeArg0)
     }
 }
+
+internal abstract class AbstractToSpecificType : AbstractInterpreter<PluginDataFrameSchema>() {
+    val Arguments.functionCall: FirFunctionCall by arg(lens = Interpreter.Id)
+    val Arguments.receiver: ConvertApproximation by arg()
+
+    override fun Arguments.interpret(): PluginDataFrameSchema {
+        val converterAnnotation = functionCall.calleeReference.toResolvedFunctionSymbol()?.getAnnotationByClassId(Names.CONVERTER_ANNOTATION, session)
+        val to = converterAnnotation?.getKClassArgument(Name.identifier("klass"), session)
```

Comment:
is this an only place where we could extract info from annotations?

Reply:
for now it's the first interpreter that relies on annotations - to reduce number of classes. imagine 20 interpreters for each to* function instead of these 3